Extending/Packages/Packages-on-Umbraco-Cloud/index.md

```diff
@@ -0,0 +1,271 @@
+---
+versionFrom: 8.0.0
+---
+
+# Packages on Umbraco Cloud
+
+If you want to use or develop packages for Umbraco Cloud there are a few things to consider and be aware of. One such thing is that custom property editors may need a ValueConnector to transform their content between environments.
+
+# ValueConnectors
+
+A ValueConnector is an extension to Deploy that allows you to transform data when you deploy content of any kind between environments. This is mostly used to transfer id based content between environments.
+
+An example of creating one in your package would be if you for example had a custom property editor that allowed you to write in the id of a media node. Not a very usable property editor but it will work for this example.
+
+So you have a property editor with a textarea input, that saves an id as a string, looks like this:
+
+![Property editor](images/property-editor.png)
+
+Then in the template you have something like this:
+
+```csharp
+<img src="@Umbraco.Media(Model.Value("BadMedia")).Url" />
+```
+
+Renders the image perfectly! 
+
+However now you do a content transfer to your Cloud environment, and one of three things will happen:
+
+1. You got super lucky and the ID you had on local happened to be the same as what your media item was assigned on your Cloud environment.
+1. Your page will now show a different image as the id you had corresponds to something else on this environment.
+1. You will get a YSOD as it can't find any media items with that id.
+
+To prevent this from happening we will need to use a ValueConnector.
+
+## Testing a ValueConnector
+
+Before we start working on making a ValueConnector a few notes on how to test and work with them. You probably will need to test the values that are being converted, but you probably also doesn't want to build, git push, content transfer to see that the value may not have changed.
```

Comment:
I'm trying to say that testing it by deploying to Cloud is not good, and then I go on to explain how it can be set up locally for faster testing. Any suggestions on how it can be reworded for clarity? 🙂

Reply:
It makes sense now. When I was reading it, I didn't realize the next part was the testing part. No need to re-write 💪